### ISSUE TITLE
Cluster layer goes blank

### ISSUE THREAD

**Issue by MarshallChen** [keplergl/kepler.gl#524]
**Describe the bug**
As said the `Cluster Layer` goes blank when zoom out at small zoom level, the official demo can reproduce. 

Change the `earthquakes` demo point layer to cluster
![image](https://user-images.githubusercontent.com/1115805/58094923-5e944700-7c04-11e9-9e4f-bb81d77f68ca.png)

 Then zoom out.
![image](https://user-images.githubusercontent.com/1115805/58094946-6d7af980-7c04-11e9-9163-e2a869b9dace.png)

**Comment by MarshallChen** [comment]
Figure out that ClusterLayers's  setState can avoid it, but I don't think it's the right way. Any help?
```diff
-    this.setState({clusters});
+    if (clusters !== null && clusters.length) {
+      this.setState({clusters});
+    }

```

**Comment by heshan0131** [comment]
hmm, it doesn't seem to happen here,  are you on the latest kepler.gl?
<img width="1394" alt="Screen Shot 2019-05-27 at 7 19 59 PM" src="https://user-images.githubusercontent.com/3605556/58446178-87a95180-80b4-11e9-9784-8741fc1c43ce.png">

**Comment by MarshallChen** [comment]
Yes the version number in screenshot is `1.0.0-2`. which is https://kepler.gl/demo/earthquakes.
As I said you have to change it to **Cluster** layer. Layer in your capture is Point layer am I right?

**Comment by MarshallChen** [comment]
![image](https://media.giphy.com/media/U3ms0LiYwWcpRrPVjZ/giphy.gif)
I made a gif

**Comment by heshan0131** [comment]
My bad, you were talking about the cluster layer, yea I was able to reproduce this, will investigate

**Comment by heshan0131** [comment]
Looks like this is inherited from the clustering algorithm in [supercluster](https://github.com/mapbox/supercluster) When zoom is at very low zoom level <4, and radius is relatively small, supercluster will return in 0 clusters

**Comment by MarshallChen** [comment]
But supercluster official demo with Leaflet works well at very low zoom level.
![image](https://user-images.githubusercontent.com/1115805/59582029-75499300-9109-11e9-8f4e-2338b1f93cb2.png)

**Comment by macrigiuseppe** [comment]
@akre54 Can you look into this please?

Thank you

**Comment by akre54** [comment]
TLDR: https://github.com/mapbox/supercluster/issues/86

----

Ok I tracked down the issue. At low zoom levels it's possible for `geoViewport.bounds` to return a minimum longitude that is < -180 or > 180, resulting in a bounding box where the minimum is larger than the maximum.

So for example:
```
longitude: -109.6070260234451
latitude: 30.862131808723703
zoom: 4
width: 1680
height: 851

const bbox = geoViewport.bounds([longitude, latitude], zoom, [width, height])
// [-183.427734375, -4.959615024698014, -35.771484375, 57.06463027327855]
// note the -183 longitude
```

A simple fix for this is to clamp the longitude to between -180 and 180, as [Supercluster already does](https://github.com/mapbox/supercluster/blob/v3.0.3/index.js#L77) for latitudes <s>(not sure why longitudes don't get this fix)</s>.

**Update: it looks like they already fixed this: https://github.com/mapbox/supercluster/issues/86. Will bump Kepler's package version.**

IMO this is a bug in `geoViewport.getBounds` since it is an invalid coordinate, but this fix should work for now.
